File: leetcode3/염혜정/4010. Maximize Pair Strength Using GCD.java
// O(n^2)

class Solution {
    public long maxPairStrength(int[] nums) {
        long max = 0;
        for (int i = 0; i<nums.length-1; i++) {
            for (int k = i+1; k<nums.length; k++) {
                long gcd = gcd(nums[i], nums[k]);
                long cal = ((long) nums[i] * nums[k]) / (gcd * gcd);
                max = Math.max(max, cal);
            }
        }
        return max;
    }

    public long gcd(long a, long b) {
        while (b>0) {
            long temp = b;
            b = a%b;
            a = temp;
        }
        return a;
    }
}

File: leetcode3/남효정/4010. Maximize Pair Strength Using GCD.py
import math

# 모든 조합으로 전부 구해본다
class Solution:
    def maxPairStrength(self, nums: list[int]) -> int:
        ans = 0

        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                a = nums[i]
                b = nums[j]

                g = math.gcd(a, b)
                strength = (a * b) // (g * g)
                ans = max(strength, ans)

        return ans   